include/nbl/core/containers/LRUCache.h

```diff
@@ -38,6 +38,13 @@ class LRUCacheBase
 		LRUCacheBase(const uint32_t capacity, MapHash&& _hash, MapEquals&& _equals, disposal_func_t&& df) : m_list(capacity, std::move(df)), m_hash(std::move(_hash)), m_equals(std::move(_equals)), searchedKey(nullptr)
 		{ }
 
+		// CAREFUL: searchedKey was left as nullptr because for ResizableLRUCache, it's always set by some method before being used
+		// If you implement another LRUCache where this is not the case and the state of the copy has to match exactly, it needs to be consistent.
+		// What I would do in such a case (either in this same class or dividing this class again into two, one that leaves the searchedKey null and 
+		// another that sets it to a consistent value) would be to iterate over every Key-Value pair in the copy, and set searchedKey to point
+		// to the Key such that `MapEquals(Key, *other.searchedKey)`
+		LRUCacheBase(const LRUCacheBase& other) : m_list(other.m_list), m_hash(other.m_hash), m_equals(other.m_equals), searchedKey(nullptr) {}
```

Comment:
searchedKey is fine to be left as nullptr, it's just a helper when `m_shortcut_map.find(invalid_iterator);` is used which leads to getReference here.
I think the whole wrapHash stuff is to avoid using `unordered_map` of Key->nodeAddr and avoid storing Key in another place.
instead we use an unordered_set, and when we want to get key associated with a `nodeAddr` we look up in our doublyLinkedList